docs/build/spells.md

```diff
@@ -0,0 +1,128 @@
+    - when to run the spell,
+    - when to unsubscribe the spell from timer events,
+    - with what period to run the spell.
+- Connection Triggers: run spells on connection/disconnection of some peers to the peer.
+- All other trigger types can be implemented based on the timer triggers & the spell KV storage (blockchain event trigger, database event trigger, HTTP event trigger, etc)
+
+## Creating and updating spells
+
+Spells can be created and managed with Fluence CLI. 
+
+Let's create a new spell in a recently scaffolded Fluence CLI project by running `fluence spell new`:
+```bash
+quickstart $ fluence spell new
+? Enter spell name 5min_spell
+Successfully generated template for new spell at /Users/arete/Documents/fluence/quickstart/src/spells/1min_spell
+? Do you want to add spell 5min_spell to a default worker defaultWorker Yes
+Added 1min_spell to defaultWorker
+```
+
+Lets check what have been generated:
+- a new spell directory was created under src/spells -- 5min_spell:
+```bash
+.
+└── 1min_spell
+    ├── spell.aqua
+    └── spell.yaml
+```
+
+It contains aqua script that will be executed and the spell.yaml file that specifies the trigger configuration for the spell.
+
+If we take a look at the spell.aqua file we see the `spell()` function that is executed when the spell is triggered:
+```aqua
+import Op, Debug from "@fluencelabs/aqua-lib/builtin.aqua"
+import Spell from "@fluencelabs/spell/spell_service.aqua"
+
+func spell():
+    msg = "Spell is working!"
+    str <- Debug.stringify(msg)
+    Spell "worker-spell"
+    Spell.list_push_string("logs", str)
+```
+
+Then lets inspect the spell.yaml file:
+```yaml
+# yaml-language-server: $schema=../../../.fluence/schemas/spell.json
+
+# Defines a spell. You can use `fluence spell new` command to generate a template for new spell
+
+# Documentation: https://github.com/fluencelabs/cli/tree/main/docs/configs/spell.md
+
+version: 0
+
+aquaFilePath: ./spell.aqua
+
+function: spell
+
+clock:
+  periodSec: 60
+  endDelaySec: 1800
+```
+
+It contains the definition of the spell, the name of function that is triggered (in case if you don't like the default name), and the cron-ish timer based trigger configuration.
+
+`periodSec` - How often the spell will be executed. If set to 0, the spell will be executed only once. If this value not provided at all - the spell will never be executed.
+
+and `endDelaySec` How long to wait before the last execution in seconds. If this property or endTimestamp not specified, periodic execution will never end. WARNING! Currently your computer's clock is used to determine a final timestamp that is sent to the server. If it is in the past at the moment of spell creation - the spell will never be executed. This property conflicts with endTimestamp. You can specify only one of them.
+
+More on the spell.yaml contents can be found in:
+Documentation: https://github.com/fluencelabs/cli/tree/main/docs/configs/spell.md 
+
+## Deploying the spell
```

Comment:
Links?

Reply:
Could you please specify what do you mean? Links to what? Or the idea is to format the headers to be linkable?

Reply:
To the deploying process doc. Spell deployment doesn't differ from a common project deployment.